File: python/src/maxSubArray.py
# https://leetcode.com/problems/maximum-subarray

from typing import List


class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_sub = nums[0]
        cur_sum = 0

        for num in nums:
            cur_sum = max(cur_sum, 0)
            cur_sum += num
            max_sub = max(max_sub, cur_sum)
        return max_sub

File: go/maxSubArray.go
// https://leetcode.com/problems/maximum-subarray

package leetcode

func maxSubArray(nums []int) int {
	res := nums[0]
	sum := 0
	for _, num := range nums {
		sum = max(sum, 0)
		sum += num
		res = max(res, sum)
	}
	return res
}
